src/consensus/pbft_manager.cpp

```diff
@@ -239,16 +239,20 @@ void PbftManager::setPbftStep(size_t const pbft_step) {
   db_->savePbftMgrField(PbftMgrRoundStep::PbftStep, pbft_step);
   step_ = pbft_step;
 
-  // if (step_ > MAX_STEPS) {
-  //   // Note: We calculate the lambda for a step independently of prior steps
-  //   //       in case missed earlier steps.
-  //   // LAMBDA_ms = 100 * LAMBDA_ms_MIN;
-  //   // LOG(log_nf_) << "Surpassed max steps, relaxing lambda to " << LAMBDA_ms
-  //   //             << " ms in round " << getPbftRound() << ", step " << step_;
-  //   LAMBDA_ms = LAMBDA_ms_MIN;
-  // } else {
-  //   LAMBDA_ms = LAMBDA_ms_MIN;
-  // }
+  if (step_ > MAX_STEPS && LAMBDA_backoff_multiple < 8) {
+    // Note: We calculate the lambda for a step independently of prior steps
+    //       in case missed earlier steps.
+    std::default_random_engine generator;
+    std::uniform_int_distribution<u_long> distribution(0, step_ - MAX_STEPS);
+    auto lambda_random_count = distribution(generator);
+    LAMBDA_backoff_multiple = 2 * LAMBDA_backoff_multiple;
+    LAMBDA_ms = LAMBDA_ms_MIN * (LAMBDA_backoff_multiple + lambda_random_count);
```

Comment:
Every time distribution will generate a number between [0, step_ - MAX_STEPS]. Looks like that cannot guaranty all nodes have same LAMBDA_ms in same steps.
Sorry, I don't understand the algorithm here

Reply:
Don't think it truly matters that nodes all have the same LAMBDA_ms once they get into the next voting steps, once they reset to next round they will al resume same lambda for the critical propose, soft vote and cert vote steps.   

The concept for adding jitter to binary exponential backoff algorithm (as I understand it) is to spread out the traffic so as to reduce the spikes in traffic in a network constrained situation in a distributed uncoordinate system. (see: https://devopedia.org/images/article/95/9334.1547359774.jpg )